### ISSUE TITLE
`fs.exists` causes a code-breaking hang.

### ISSUE THREAD

**Issue by crenwick** [koajs/send#24]
Took me a while to find the source of this. 
In the context of `koa-static`, I have some routes that looks like this:
```javascript
app.use(route.post('/upload', routes.upload));
// ... other routes here
app.use(serve(__dirname + '/build'));
```
Which works great at first and the serve route serves my build directory statically.
However, once the post route is successfully used once, the `serve` generator never returns if it is called again. Instead the debug outputs:
```javascript
 <-- GET /
  koa-send send "/" {"root":".../build","index":"index.html"} +2m
```
but it never bubbles back up the middleware chain.

After digging into the source, I found that it's hanging on [this line in koa-send](https://github.com/koajs/send/blob/master/index.js#L64). 
Breaking this apart more, I found that whatever happens after a file upload route is used causes `yield fs.exists(path + '.gz')` to hang and never return, which blocks `koa-send` from returning.

I frankensteined this block code to use `fs.exists` as a callback (ie, not the `mz/fs` wrapper), and that works fine.

I also updated `koa-send`'s `mz/fs` dependency to 2.0.0, but it didn't solve the issue.

*Notes:*
- All other routes continue to work fine, as they don't rely on `koa-send`
- I changed the router and upload route a few times, to use different styles of handling the upload with different dependencies. All of them cause this problem.
- I also changed the `koa-static` route to serve files in different ways and with different dependencies, but all the effective ways of serving static files rely on `koa-send`.
- I can see this being an issue to report with `mz/fs`, but I want to report it here first as this code should not block server routes.

**Comment by jonathanong** [comment]
weird. anyone know what's going on here?

**Comment by crenwick** [comment]
Forgot I had this open.

I figured out the problem a while back. I was using a library that used `co-fs`, and both `co-fs` and and `mz/fs` were replacing the node-standard behavior of the built in fs library.

I solved the issue by forking `koa-send` and changing [this line](https://github.com/koajs/send/blob/master/index.js#L13) to something like 
`var mzfs = require('mz/fs');` (and changing all the internal references).

Though this issue can now be closed, I would highly recommend making the above change to this library so that it doesn't continue to have this behavior for unsuspecting devs. It was a pain in the ass the pin down 😅

I'd be more than happy to make the PR if it is something this team is interested in.
